---
version: '3'
services:
  portainer:
    container_name: 'portainer'
    image: 'portainer/portainer-ce:latest'
    command: '-H tcp://socket-proxy:2375'
    restart: '${RESTARTAPP}'
    networks:
      - proxy
      - socket-proxy
    security_opt:
      - 'no-new-privileges:true'
    volumes:
      - '${APPFOLDER}/portainer:/data'
    environment:
      - 'PGID=${ID}'
      - 'PUID=${ID}'
      - 'TZ=${TZ}'
    labels:
      - 'dockupdater.enable=true'
      - 'traefik.enable=true'
      - 'traefik.docker.network=proxy'
      - 'traefik.http.routers.frontend.rule=Host(`portainer.${DOMAIN}`)'
      - 'traefik.http.routers.frontend.entrypoints=https'
      - 'traefik.http.services.frontend.loadbalancer.server.port=9000'
      - 'traefik.http.routers.frontend.service=frontend'
      - 'traefik.http.routers.frontend.tls.certresolver=dns-cloudflare'
networks:
  proxy:
    driver: bridge
    external: true
  socket-proxy:
    driver: bridge
    external: true